server/broadcast.go

```diff
@@ -28,6 +25,8 @@ var Policy *verification.Policy
 var BroadcastCfg = &BroadcastConfig{}
 var MaxAttempts = 3
 
+var downloadSeg = drivers.GetSegmentData
```

Comment:
I don't feel great about declaring this package level variable, but this made writing tests for this PR much easier because I was able to monkey patch the `drivers.GetSegmentData` implementation.

An alternative could be extract the [dlFunc](https://github.com/livepeer/go-livepeer/blob/b6e07fee00114427cd8b7638976a53fc49bfcc54/server/broadcast.go#L382) function which is defined inline in `transcodeSegment()` into a standalone function. Then, `dlFunc` could accept a function with the signature `func(url string) ([]byte, error)` as an argument - in practice, `drivers.GetSegmentData` would be passed in, but a stub function could be used in tests. This would've required a bit of refactoring so I opted not to take this approach right now, but could be worth considering outside of this PR.

Reply:
There's a few other areas where we currently do this to make writing tests much easier. I feel this is okay for now but we should probably revisit if we can think of a pattern that doesn't require a huge refactor or add it to tech debt.